shaderc-sys/build/build.rs

```diff
@@ -0,0 +1,194 @@
+        // C/C++ flags, which causes problems. So I have to manually
+        // define CMAKE_*_FLAGS_* here to suppress that.
+        .define("CMAKE_C_FLAGS", " /nologo /EHsc")
+        .define("CMAKE_CXX_FLAGS", " /nologo /EHsc")
+        .define("CMAKE_C_FLAGS_RELEASE", " /nologo /EHsc")
+        .define("CMAKE_CXX_FLAGS_RELEASE", " /nologo /EHsc")
+        .define("CMAKE_INSTALL_LIBDIR", "lib")
+        .generator("Ninja");
+    config.build()
+}
+
+fn main() {
+    // Deprecated --no-defaults path inhereted from before shaderc-rs & shaderc-sys split
+    if env::var("CARGO_FEATURE_CHECK_PASSTHROUGH").is_ok()
+        && env::var("CARGO_FEATURE_DONT_USE_DEPRECATED").is_err()
+    {
+        let out_dir = env::var("OUT_DIR").unwrap();
+        println!("cargo:warning=USE OF --no-defaults IS DEPRECATED BEHAVIOR.");
+        println!("cargo:warning=Requested to skip building native C++ shaderc.");
+        println!(
+            "cargo:warning=Searching {} for shaderc_combined static lib...",
+            out_dir
+        );
+        println!("cargo:rustc-link-search=native={}", out_dir);
+        println!("cargo:rustc-link-lib=static=shaderc_combined");
+        emit_stdc_link();
+        return;
+    }
+
+    // Explicit overrides first
+    let mut search_dir = if let Ok(lib_dir) = env::var("SHADERC_LIB_DIR") {
+        println!("cargo:warning=Using specified pre-built libshaderc");
+        Some(lib_dir)
+    } else if let Ok(lib_path) = env::var("SHADERC_LIB_PATH") {
+        println!("cargo:warning=Using specified pre-built libshaderc");
+        Some(lib_path)
+    } else if let Ok(lib_path) = env::var("SHADERC_STATIC") {
+        println!("cargo:warning=Using specified pre-built libshaderc");
+        Some(lib_path)
+    } else {
+        None
+    };
+
+    let target_os = env::var("CARGO_CFG_TARGET_OS").unwrap();
+    let target_env = env::var("CARGO_CFG_TARGET_ENV").unwrap();
+
+    // Guess linux path for system library unless requested to do source build
+    if search_dir.is_none()
+        && target_os == "linux"
+        && env::var("CARGO_FEATURE_BUILD_FROM_SOURCE").is_err()
+    {
+        println!("cargo:warning=Checking for system installed libraries");
+        search_dir = Some("/usr/lib/".to_owned());
+    }
+
+    // Try to build with the static lib if a path was received or chosen
+    if let Some(search_dir) = search_dir {
+        let search_path = Path::new(&search_dir);
+        let combined_lib_path = search_path.join(COMBINED_LIB_FILE);
+        let dylib_name = format!("{}shaderc{}", consts::DLL_PREFIX, consts::DLL_SUFFIX);
+        let dylib_path = search_path.join(dylib_name.clone());
+
+        if let Some((lib_dir, lib_name)) = {
+            if combined_lib_path.exists() {
+                Some((search_dir.clone(), COMBINED_LIB.to_owned()))
+            } else if dylib_path.exists() {
+                Some((search_dir.clone(), dylib_name))
```

Comment:
Should we prefer dynamic lib to static lib? What do you think? But it's fine to do in another PR.

Reply:
My vantage point is from years of only maintaining Linux machines, so it's actually quite hard for me to anticipate or reason about their choices.

I cargo-culted the bits that try to build a Windows dll path from the same document I linked before.  I truly have no idea what kind of library might get built on Windows.

Reply:
You might have been asking about dynamic on Linux.  I think I read that static was more likely to succeed.  Not specifying would be cool since it will search more accurately.

The pkg-config file has landed, but it will be a little bit before it hits package managers.  As for from source builds, I believe static is default when you bring your own version because you can only share if everyone is on the same page (using a system package manager) 
https://github.com/google/shaderc/commit/439a848c0c0e91b5f1b85059cd4b059274b9a93b

Reply:
SG!